torchtune/utils/_perf_profiler.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import ContextManager, Optional
+
+import torch
+from torch.profiler import profile
+
+
+def perf_profiler(
+    output_dir: Optional[str] = "./torchtune_perf_tracing.json",
+) -> ContextManager:
+    """
+    Utility component to measure the time and memory consumption of the model’s operators.
+    check the user manual: https://pytorch.org/docs/stable/profiler.html for more details.
+    The schedule for this profiler is wait 5 steps, warmup 5 steps, trace 5 steps
```

Comment:
IMO we need to be waiting for way more steps. Torch compile and FSDP iterations take a bit longer for this to warm up. 

Some initial defaults I'd put are wait 100 steps, warm up 5, trace 5.